src/main/java/org/wildfly/security/sasl/plain/PlainSaslClientFactory.java

```diff
@@ -42,7 +42,7 @@
     public SaslClient createSaslClient(final String[] mechanisms, final String authorizationId, final String protocol, final String serverName, Map<String, ?> props, final CallbackHandler cbh) throws SaslException {
         Assert.checkNotNullParam("cbh", cbh);
         if (props == null) props = Collections.emptyMap();
-        if (PlainSasl.isMatched(props)) for (String mechanism : mechanisms) {
+        if (PlainSasl.isMatched(props, true)) for (String mechanism : mechanisms) {
```

Comment:
Looks like there are true/false for query conversely - should not be false for create and true for getMechanismNames?